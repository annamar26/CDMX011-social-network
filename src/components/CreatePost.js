import { fbFunctions } from '../firebaseClient.js';
import { onNavigate } from '../routes.js';

const CreatePost = () => {
  const conteiner = document.createElement('div');
  const html = `  <nav>
    <div>
    <img class='logo' src='./icons&img/logo.png'></div>
    <div id='profile' class='icons img-profile'>
    </div>
       </nav>
       <div class='create-post'>
    <form id='postForm'>
    <h3>Crear publicación</h3>
    <input id='autor' type=text placeholder="Autor del libro" required>
    <input id='title' type=text placeholder="Titulo del libro" required>
    <textarea id='content-input' type=text placeholder="Escribe aquí tu reseña. Cuenta tu experiencia con la lectura del libro que desees" maxlength="1200" required></textarea>
    <button id='publicar' class='button' type=submit>Publicar</button>
    </form>
    </div>`;

  conteiner.innerHTML = html;
  fbFunctions.setWelcome();
  fbFunctions.comprobar();
  const postCreated = conteiner.querySelector('#postForm');
  postCreated.addEventListener('submit', (e) => {
    e.preventDefault();
    const bookAutor = conteiner.querySelector('#autor').value;
    const bookTitle = conteiner.querySelector('#title').value;
    const postContent = conteiner.querySelector('#content-input').value;

    fbFunctions.createPost(bookAutor, bookTitle, postContent)
      .then(() => {
        console.log('Document successfully written!');
        fbFunctions.getPosts();
        onNavigate('/timeline');
      })
      .catch((error) => {
        console.error('Error writing document: ', error);
      });
  });
  return conteiner;
};

export default CreatePost;
