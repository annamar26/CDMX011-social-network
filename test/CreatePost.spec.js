/**
 * @jest-environment jsdom
 */
import CreatePost from '../src/components/CreatePost.js';
import { render } from '../src/utils.js';

describe('Create Post', () => {
  document.body.innerHTML = '<div id="root"></div>';
  const rootDiv = document.getElementById('root');
  it('should render Create Post page', () => {
    render(rootDiv, CreatePost());

    expect(rootDiv).toMatchSnapshot();
  });
});
