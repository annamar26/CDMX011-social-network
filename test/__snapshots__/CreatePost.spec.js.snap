// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Post should render Create Post page 1`] = `
"() =&gt; {
  /* istanbul ignore next */
  cov_2756w4ghya().f[0]++;
  const conteiner =
  /* istanbul ignore next */
  (cov_2756w4ghya().s[1]++, document.createElement('div'));
  const html =
  /* istanbul ignore next */
  (cov_2756w4ghya().s[2]++, \`  &lt;nav&gt;
    &lt;div&gt;
    &lt;img class='logo' src='./icons&amp;img/logo.png'&gt;&lt;/div&gt;
    &lt;div id='profile' class='icons img-profile'&gt;
    &lt;/div&gt;
       &lt;/nav&gt;
       &lt;div class='create-post'&gt;
    &lt;form id='postForm'&gt;
    &lt;h3&gt;Crear publicación&lt;/h3&gt;
    &lt;input id='autor' type=text placeholder=\\"Autor del libro\\" required&gt;
    &lt;input id='title' type=text placeholder=\\"Titulo del libro\\" required&gt;
    &lt;textarea id='content-input' type=text placeholder=\\"Escribe aqií tu reseña\\" maxlength=\\"1200\\" required&gt;&lt;/textarea&gt;
    &lt;button id='publicar' class='button' type=submit&gt;Publicar&lt;/button&gt;
    &lt;/form&gt;
    &lt;/div&gt;\`);

  /* istanbul ignore next */
  cov_2756w4ghya().s[3]++;
  conteiner.innerHTML = html;

  /* istanbul ignore next */
  cov_2756w4ghya().s[4]++;

  /* istanbul ignore next */
  _firebaseClient.
  /* istanbul ignore next */
  fbFunctions.setWelcome();

  /* istanbul ignore next */
  cov_2756w4ghya().s[5]++;

  /* istanbul ignore next */
  _firebaseClient.
  /* istanbul ignore next */
  fbFunctions.comprobar();

  const postCreated =
  /* istanbul ignore next */
  (cov_2756w4ghya().s[6]++, conteiner.querySelector('#postForm'));

  /* istanbul ignore next */
  cov_2756w4ghya().s[7]++;
  postCreated.addEventListener('submit', (e) =&gt; {
    /* istanbul ignore next */
    cov_2756w4ghya().f[1]++;
    cov_2756w4ghya().s[8]++;
    e.preventDefault();
    const bookAutor =
    /* istanbul ignore next */
    (cov_2756w4ghya().s[9]++, conteiner.querySelector('#autor').value);
    const bookTitle =
    /* istanbul ignore next */
    (cov_2756w4ghya().s[10]++, conteiner.querySelector('#title').value);
    const postContent =
    /* istanbul ignore next */
    (cov_2756w4ghya().s[11]++, conteiner.querySelector('#content-input').value);

    /* istanbul ignore next */
    cov_2756w4ghya().s[12]++;

    /* istanbul ignore next */
    _firebaseClient.
    /* istanbul ignore next */
    fbFunctions.createPost(bookAutor, bookTitle, postContent).then(() =&gt; {
      /* istanbul ignore next */
      cov_2756w4ghya().f[2]++;
      cov_2756w4ghya().s[13]++;
      console.log('Document successfully written!');

      /* istanbul ignore next */
      cov_2756w4ghya().s[14]++;

      /* istanbul ignore next */
      _firebaseClient.
      /* istanbul ignore next */
      fbFunctions.getPosts();

      /* istanbul ignore next */
      cov_2756w4ghya().s[15]++;
      onNavigate('/timeline');
    }).catch((error) =&gt; {
      /* istanbul ignore next */
      cov_2756w4ghya().f[3]++;
      cov_2756w4ghya().s[16]++;
      console.error('Error writing document: ', error);
    });
  });

  /* istanbul ignore next */
  cov_2756w4ghya().s[17]++;
  return conteiner;
}"
`;
